lib/android/src/main/java/com/airbnb/android/react/maps/AirMapMarkerManager.java

```diff
@@ -24,7 +31,69 @@
   private static final int HIDE_INFO_WINDOW = 2;
   private static final int ANIMATE_MARKER_TO_COORDINATE = 3;
 
+
+  private ConcurrentHashMap<String, BitmapDescriptor> markerIconBitmapDescriptorHolder;
+  private ConcurrentHashMap<String, Bitmap> markerIconBitmapHolder;
+  private ConcurrentHashMap<String, BitmapDescriptor> markerTextedIconBitmapDescriptorHolder;
+
   public AirMapMarkerManager() {
+    markerIconBitmapDescriptorHolder = new ConcurrentHashMap<>();
+    markerIconBitmapHolder = new ConcurrentHashMap<>();
+    markerTextedIconBitmapDescriptorHolder = new ConcurrentHashMap<>();
+  }
+
+  public void saveMarkerBitmapDescriptorFor(String uri, BitmapDescriptor bitmapDescriptor) {
+    if(uri == null) { return; }
+    this.markerIconBitmapDescriptorHolder.put(uri, bitmapDescriptor);
+  }
+
+  public void saveMarkerIconBitmapFor(String uri, Bitmap bitmap){
+    if(uri == null) { return; }
+    this.markerIconBitmapHolder.put(uri, bitmap);
+  }
+
+  private void saveMarkerTextIconBitmapDescriptorFor(String uri, String markerText, BitmapDescriptor bitmapDescriptor){
+    if(uri == null && markerText == null) { return; }
+    String key = this.generateMarkerTextKey(uri, markerText);
+    this.markerTextedIconBitmapDescriptorHolder.put(key, bitmapDescriptor);
+  }
+
+  public BitmapDescriptor getMarkerBitmapDescriptorFor(String uri) {
+    if(uri == null) { return null; }
+    return this.markerIconBitmapDescriptorHolder.get(uri);
+  }
+
+  public Bitmap getMarkerIconBitmapFor(String uri) {
+    if(uri == null) { return null; }
+    return this.markerIconBitmapHolder.get(uri);
+  }
+
+  public synchronized BitmapDescriptor getMarkerTextBitmapDescriptorFor(String uri, String markerText) {
+    if(uri == null && markerText == null) { return null; }
+    String key = this.generateMarkerTextKey(uri, markerText);
+    BitmapDescriptor result = this.markerTextedIconBitmapDescriptorHolder.get(key);
+    if(result != null) {return result;}
+    Bitmap bitmap = this.getMarkerIconBitmapFor(uri);
+    if(bitmap == null) { return null; }
+    result = this.drawTextOnBitmap(markerText, bitmap);
+    this.saveMarkerTextIconBitmapDescriptorFor(uri, markerText, result);
```

Comment:
Could you space things a bit so it's easier to read?

Reply:
I will do a refactor of the code.

Reply:
Hi kilcaz, I have updated the code. This pr will not be merged, it is created just to show what has been updated.

Reply:
Sure, but if the code is good and optimizes the use of this library we should open a PR into the main repo